core/domain/rights_manager.py

```diff
@@ -1007,6 +1146,9 @@ def _release_ownership_of_activity(committer, activity_id, activity_type):
         raise Exception(
             'The ownership of this %s cannot be released.' % activity_type)
 
+    # Ruling out the possibility of None for mypy type checking.
+    assert activity_rights is not None
```

Comment:
Same question about guarantees.

Reply:
Similarly, we are assessing attributes without checking for none cases.

Reply:
Nope, see above.

Reply:
Here, we are passing `activity_rights` to `check_can_release_ownership` method in line `1167`. If `activity_rights` is None then `check_can_release_ownership` is going to return false and an exception is raised in line 1167.

Reply:
https://github.com/oppia/oppia/blob/0e69c2bcd7a91ef3595da274d060608efd3140ae/core/domain/rights_manager.py#L684

Reply:
This is a good justification and it works fine for this case. Thanks! The comment is also great.

However, it would probably be better to have an "if activity_rights is None" clause just after line 1165 with a more specific error message. Then the assertion makes sense in the context of this function and will not change if the logic in check_can_release_ownership is changed in the future.

Reply:
"strict" paradigm is added to `_get_activity_rights` and `strict=True` used here, Now `_get_activity_rights` will throw it's own error ( if activity_rights is not found for the given activity_id ).

Reply:
This works, but you'll need to update the docstring to cover the new Exception.

Reply:
Done!